connect/api/src/test/java/org/apache/kafka/connect/storage/NumberConverterTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.storage;
+
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public abstract class NumberConverterTest<T extends Number> {
+    private static final String TOPIC = "topic";
+    private static final String HEADER_NAME = "header";
+
+    private T[] samples;
+    private Schema schema;
+    private NumberConverter<T> converter;
+    private Serializer<T> serializer;
+
+    protected abstract T[] samples();
+
+    protected abstract NumberConverter<T> createConverter();
+
+    protected abstract Serializer<T> createSerializer();
+
+    protected abstract Schema schema();
+
+    @Before
+    public void setup() {
```

Comment:
minor nit : why not use @parameterized and make schema, converter, serializer, etc part of parameters?